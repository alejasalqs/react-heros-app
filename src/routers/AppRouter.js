import React, { useContext } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { AuthContext } from "../auth/AuthContext";
import { LoginScreen } from "../components/login/LoginScreen";
import { DashboardRoutes } from "./DashboardRoutes";
import { PrivateRoutes } from "./PrivateRoutes";
import { PublicRoutes } from "./PublicRoutes";
// Sistema de rutas principal de la aplicacion
export const AppRouter = () => {
  const { user } = useContext(AuthContext);
  return (
    <Router>
      <>
        <Switch>
          <PublicRoutes
            exact
            path="/login"
            component={LoginScreen}
            isAuthenticated={user.logged}
          />

          {/*En este caso no es necesario el exact por las rutas que estan abajo de esta*/}
          <PrivateRoutes
            path="/"
            component={DashboardRoutes}
            isAuthenticated={user.logged}
          />
        </Switch>
      </>
    </Router>
  );
};
