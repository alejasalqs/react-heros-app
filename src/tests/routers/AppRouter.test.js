import React from "react";
import { shallow } from "enzyme";
import { AppRouter } from "../../routers/AppRouter";
import { AuthContext } from "../../auth/AuthContext";

describe("Pruebas en el componente <AppRouter/>", () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false,
    },
  };
  test("Debe de mostrar el login si no esta autenticado ", () => {
    const wrapper = shallow(
      // Aqui es mejor usar mount
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar el componente de marvel si estÃ¡ autenticado ", () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        name: "Alejandro",
        logged: true,
      },
    };
    const wrapper = shallow(
      // Aqui es mejor usar mount
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    // Esta prueba falla porque no puede ver dentro del wrapper el navbar
    expect(wrapper.find(".navbar").exists()).toBe(true);
  });
});
