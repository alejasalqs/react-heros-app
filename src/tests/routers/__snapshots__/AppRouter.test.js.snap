// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AppRouter/> Debe de mostrar el login si no esta autenticado  1`] = `<AppRouter />`;
